TOOLS/ce/src/main/scala/org.opalj.ce/CommentParser.scala

```diff
@@ -0,0 +1,308 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package ce
+
+import java.nio.file.Path
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.io.Source
+import scala.util.Using
+import scala.util.control.Breaks.break
+import scala.util.control.Breaks.breakable
+
+/**
+ * The Comment Parser is responsible for parsing the HOCON files and its comments
+ * Do NOT use this class directly, use ConfigParserWrapper for safe parsing instead
+ */
+class CommentParser() {
+    var iterator: Iterator[String] = null
```

Comment:
So, here is the issue. Since the entire CommentParser structure jointly accesses the Iterator to parse a file, using it as a global variable guarantees it having a synchronized state for all methods. 
Reaching a null pointer exception is impossible, because the only public method of the class sets the iterator. 
I also made the iterator private in the meantime, because it is not supposed to be accessed by anything else than the CommentParsers own methods.
I think that actually passing it to the methods as parameter will actually have a negative effect on stability, as it may enable the iterator to become async at some point during the parsing process. 
Using it as a parameter is also not a great idea, because it is used by all but two methods in the class.

Reply:
I am initializing the iterator as an empty iterator now. Should someone manage to call one of the internal persing methods, all that happens is that an empty structurre is returned (which is a correct behavious) instead of a null pointer exception.